framework/src/functions/PiecewiseTabularBase.C

```diff
@@ -17,9 +17,17 @@ PiecewiseTabularBase::validParams()
 {
   InputParameters params = PiecewiseBase::validParams();
 
-  MooseEnum axis("x=0 y=1 z=2");
-  params.addParam<MooseEnum>(
-      "axis", axis, "The axis used (x, y, or z) if this is to be a function of position");
+  MooseEnum axis("x=0 y=1 z=2 t=3");
+  params.addDeprecatedParam<MooseEnum>(
+      "component",
+      axis,
+      "The axis used (x, y, or z) if this is to be a function of position.",
+      "This option is a deprecated alias for axis, use axis instead.");
+  params.addParam<MooseEnum>("axis",
+                             axis,
+                             "The axis used (x, y, or z) if this is to be a function of position. "
+                             "If this is to be a time-dependent function, t may also be specified "
+                             "or omitted. ");
```

Comment:
What's your rationality for switching all of this?

I'm not completely against it, but we need to rework the logic here some.

Reply:
I want to preserve functionality, so this is my naïve attempt at ensuring that `t` can be specified from `SplineFunction`, while also preserving the default component of `SplineFunction` (which is `x`). I'm also concerned with this that I may break something while overriding the default value in `SplineFunction::validParams()`

Reply:
As for why I'm modifying the parent class, it was a suggestion from Guillaume on the moose slack channel, so that the base class could handle the previously-undefined-now-deprecated input.